### PR TITLE
refactor: Moving from jest to vitest

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Moving unit testing framework from Jest to Vitest in order to speed things up.

Going from this:
<kbd></kbd>

To this:
<kbd></kbd>

Marking a 85.3% decrease in time, even running more tests than Jest.

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

### CODE REVIEW

**Comment by zomars** [comment]
I'm loving this! Nice job @leog 

Just for completeness sake. What are the drawbacks of abandoning jest in our codebase?

File: .prettierignore [hunk 1]
```diff
@@ -7,6 +7,7 @@ public
 
 *.lock
 *.log
+*.test.ts
```
Comment (zomars) [inline .prettierignore:10]: Why is this needed?

File: apps/web/test/lib/getTimezone.test.ts [hunk 2]
```diff
@@ -34,25 +34,25 @@ const option = {
 };
 
 it("should return empty array for an empty string", () => {
-  expect(filterByCities("", cityData)).toMatchInlineSnapshot(`Array []`);
+  expect(filterByCities("", cityData)).toMatchInlineSnapshot(`[]`);
 });
 
 it("should filter cities for a valid city name", () => {
   expect(filterByCities("San Francisco", cityData)).toMatchInlineSnapshot(`
-    Array [
```
Comment (hariombalhara) [inline apps/web/test/lib/getTimezone.test.ts:42]: I had always thought why it was needed. Now, vitest removing it ❤️

File: .vscode/launch.json [hunk 3]
```diff
@@ -5,11 +5,12 @@
       "type": "node",
       "request": "launch",
       "name": "Next.js Node Debug",
-      "runtimeExecutable": "${workspaceFolder}/node_modules/next/dist/bin/next",
+      "runtimeExecutable": "/Users/leog/.yarn/bin/yarn",
```
Comment (hariombalhara) [inline .vscode/launch.json:8]: This need to be generic.

File: .vscode/launch.json [hunk 4]
```diff
@@ -5,11 +5,12 @@
...
+      "runtimeArgs": ["vitest", "weekday.test.ts"],
```
Comment (hariombalhara) [inline .vscode/launch.json:9]: This shouldn't be commited I think

File: apps/web/pages/api/intercom-hash.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
+import crypto from "crypto";
```
Comment (hariombalhara) [inline apps/web/pages/api/intercom-hash.ts:1]: I am not sure if this is intentional.

File: vitest.config.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
+  test: {
+    coverage: {
+      provider: "c8",
```
Comment (hariombalhara) [inline vitest.config.ts:6]: Will the coverage data be visible in CI ?

**Review by hariombalhara — APPROVED** [review]
LGTM ! Awesome work @leog

**Comment by PeerRich** [comment]
happy to merge once open questions were answered

File: .vscode/launch.json (hunk 3 above)
Comment (leog) [reply .vscode/launch.json:8]: Whoops, yeah, slipped!

**Comment by leog** [comment]
> Just for completeness sake. What are the drawbacks of abandoning jest in our codebase?

@zomars There is no apparent drawback other than immersing ourselves into a new framework which can lead to unknowns although Vitest is being used more and more these days. Not to mention Vitest has been designed with a Jest compatible API. More info about migration and similarities/differences [here](https://vitest.dev/guide/migration.html).

All in all, I think we should give it a chance, the time difference is huge, and locally is ludicrous, see: 
image

**Review by emrysal — APPROVED** [review]

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 7]
```diff
@@ -488,7 +488,7 @@ function getBookingData({
             }
           }
         });
-
+  debugger;
```
Comment (emrysal) [inline packages/features/bookings/lib/handleNewBooking.ts:491]: BIG OOPS.
